### ISSUE TITLE
Update request: r8125 9.013.02 → 9.014.01

### ISSUE THREAD

**Issue by MakiseKurisu** [NixOS/nixpkgs#361324]
## Package Information

- Package name: r8125
- Latest released version: 9.014.01
- Current version on the unstable channel: 9.013.02
- Current version on the stable/release channel: 9.013.02

## Checklist

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

## Notify maintainers

@notpeelz

---

Note for maintainers: Please tag this issue in your PR.

Currently the 9.013.02 build fails on 6.12.1:

```
       >   CC [M]  /build/source/src/r8125_n.o
       > /build/source/src/r8125_n.c:7682:20: error: initialization of 'int (*)(struct net_device *, struct ethtool_keee *)' from incompatible pointer type 'int (*)(struct net_device *, struct ethtool_eee *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
       >  7682 |         .get_eee = rtl_ethtool_get_eee,
       >       |                    ^~~~~~~~~~~~~~~~~~~
       > /build/source/src/r8125_n.c:7682:20: note: (near initialization for 'rtl8125_ethtool_ops.get_eee')
       > /build/source/src/r8125_n.c:7683:20: error: initialization of 'int (*)(struct net_device *, struct ethtool_keee *)' from incompatible pointer type 'int (*)(struct net_device *, struct ethtool_eee *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
       >  7683 |         .set_eee = rtl_ethtool_set_eee,
       >       |                    ^~~~~~~~~~~~~~~~~~~
       > /build/source/src/r8125_n.c:7683:20: note: (near initialization for 'rtl8125_ethtool_ops.set_eee')
```

Also, I wonder if it is possible to ease or update the broken restriction. I'm currently having issues with r8169 driver, and currently the workaround is to ssh from another NIC and `rmmod r8169 && modprobe r8169`. Even this would fail after while when the device locks up some resources that prevents driver from working, and at that point I have to reboot the machine. So if I can use this driver without specifying `nixpkgs.config.allowBroken = true;` that can be very helpful to check this type of issues.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by MakiseKurisu** [comment]
Ummm maybe my issue is fixed in [6.12](https://github.com/torvalds/linux/commit/b9c7ac4fe22c608acf6153a3329df2b6b6cd416c). Gonna try that kernel without r8125. I was on 6.11 with 24.05.

**Comment by notpeelz** [comment]
I don't use NixOS anymore so it's a bit hard for me to test this package.

**Comment by notpeelz** [comment]
See #361615

**Comment by MakiseKurisu** [comment]
Fixed by #361615
